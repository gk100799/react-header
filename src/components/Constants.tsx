// import logo from './logo.png'

interface Tabs {
    name: string,
    path: string,
    props: CoverProps
}

export interface CoverProps {
    coverHeading?: string,
    coverContent?: string,
    buttonTitle?: string,
}

// export const logo: File = logo;
export const logo: string = "Logo";

export const tabs: Tabs[] = [
    {
        name: 'Home',
        path: '/',
        props: {
            coverHeading: 'This is Home page.',
            coverContent: ' Lorem ipsum dolor sit amet consectetur adipisicing elit. Totam magni illum molestiae reiciendis et quis libero porro neque suscipit beatae sunt facilis ullam corrupti temporibus voluptatibus, facere, quod corporis similique? ',
            buttonTitle: 'Subscribe'
        }
    },
    {
        name: 'My Account',
        path: '/account',
        props: {
            coverHeading: 'This is My Accounts page.',
            coverContent: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Accusamus mollitia, quae nihil a eius, quisquam quaerat deserunt obcaecati dolore earum ipsam ratione nobis culpa sequi! Obcaecati ab rerum alias nulla!',
            buttonTitle: 'Learn more'
        }
    }
];