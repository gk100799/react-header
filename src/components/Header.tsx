import React, { Component } from 'react'
import { NavLink } from 'react-router-dom';
import { tabs, logo } from './Constants'

class Header extends Component {
    render() {
        return (
            <header className="masthead">
                <div className="inner">
                    <NavLink to="/" exact activeClassName="none" ><h3 className="masthead-brand" data-testid="logo">{logo}</h3></NavLink>
                    <nav className="nav nav-masthead justify-content-center">
                        {tabs && tabs.map(tab => (
                            <NavLink data-testid={tab.name} key={tab.name} exact to={tab.path} activeClassName="active" className="nav-link" >{tab.name}</NavLink >
                        ))}
                    </nav>
                </div>
            </header>
        )
    }
}

export default Header