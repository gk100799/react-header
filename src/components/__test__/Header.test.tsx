import React from 'react';
import ReactDOM from 'react-dom'
import Header from '../Header'
import { BrowserRouter as Router } from 'react-router-dom'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import { tabs, logo } from '../Constants'

afterEach(cleanup);

it("renders Header without crashing", () => {
    const div = document.createElement("div")
    ReactDOM.render(<Router><Header /></Router>, div)
})

it("renders Header correctly without modifying the constants file", () => {
    // console.log(logo)
    const { getByTestId } = render(<Router><Header /></Router>)
    expect(getByTestId(tabs[0].name)).toHaveTextContent(tabs[0].name)
    expect(getByTestId("logo")).toHaveTextContent(logo)
})

it("matches snapshot", () => {
    const tree = renderer.create(<Router><Header /></Router>)
    expect(tree).toMatchSnapshot();
})