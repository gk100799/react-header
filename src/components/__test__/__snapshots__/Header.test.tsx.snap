// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<header
  className="masthead"
>
  <div
    className="inner"
  >
    <h3
      className="masthead-brand"
      data-testid="logo"
    >
      Logo
    </h3>
    <nav
      className="nav nav-masthead justify-content-center"
    >
      <a
        aria-current="page"
        className="nav-link active"
        data-testid="Home"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
      <a
        aria-current={null}
        className="nav-link"
        data-testid="My Account"
        href="/account"
        onClick={[Function]}
      >
        My Account
      </a>
    </nav>
  </div>
</header>
`;
